### ISSUE TITLE
[BUG] [aws-identity] IntelliJCredential not working with Azure Toolkit for IntelliJ

### ISSUE THREAD

**Issue by raniemi** [Azure/azure-sdk-for-java#21150]
**Describe the bug**
When trying to authenticate with Azure via IntelliJ from a Java/Sprint application, the following error is thrown:
```
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "X" (class com.azure.identity.implementation.IntelliJAuthMethodDetails), not marked as ignorable (4 known properties: "authMethod", "azureEnv", "accountEmail", "credFilePath"])
 at [Source: (File); line: 1, column: 48] (through reference chain: com.azure.identity.implementation.IntelliJAuthMethodDetails["X"])
...
at com.microsoft.graph.authentication.TokenCredentialAuthProvider.getAuthorizationTokenAsync(TokenCredentialAuthProvider.java:58)
	at com.microsoft.graph.httpcore.AuthenticationHandler.intercept(AuthenticationHandler.java:54)
```
And after systematic removal of non-ignorable fields (`tenantId`, `clientId`, `authType`) from `AuthMethodDetails.json`, you get:
```
com.azure.identity.CredentialUnavailableException: IntelliJ Authentication not available. Please login with Azure Tools for IntelliJ plugin in the IDE.
	at com.azure.identity.implementation.IdentityClient.authenticateWithIntelliJ(IdentityClient.java:384)
	at com.azure.identity.IntelliJCredential.lambda$getToken$2(IntelliJCredential.java:87)
        ...
        at com.microsoft.graph.authentication.TokenCredentialAuthProvider.getAuthorizationTokenAsync(TokenCredentialAuthProvider.java:58)
	at com.microsoft.graph.httpcore.AuthenticationHandler.intercept(AuthenticationHandler.java:54)
        ...
```
And the application halts.

**To Reproduce**
* In IntelliJ, sign into Azure using your favorite method: `Tools > Azure > Azure Sign In`
* In a terminal, run: `cp -r ~/.AzureToolsForIntelliJ ~/AzureToolsForIntelliJ` (it appears that [17499](https://github.com/Azure/azure-sdk-for-java/issues/17499) is still an issue).
* Start your Java/Spring application which creates a `IntelliJCredential` etc. (like in the code snippet)
* You will see X since the contents of `~/AzureToolsForIntelliJ/AuthMethodDetails.json` has additional fields that weren't expected
* Removing each of the unexpected fields from `~/AzureToolsForIntelliJ/AuthMethodDetails.json` will generate a final exception without any additional information what can be done to resolve it.

***Code Snippet***
```groovy
        // i.e. scopes = [ 'https://graph.microsoft.com/.default' ]
        final IntelliJCredential intelliJCredential = new IntelliJCredentialBuilder()
                .tenantId(tenantId)   // same problem with/without
                .build()

        final TokenCredentialAuthProvider tokenCredentialAuthProvider =
                new TokenCredentialAuthProvider(scopes, intelliJCredential);

        // For illustration purposes
        GraphServiceClient
                .builder()
                .authenticationProvider(tokenCredentialAuthProvider)
                .buildClient()
```

**Expected behavior**
For my Java/Spring application to be able to use the Azure Toolkit for IntelliJ to construct a usable `TokenCredentialAuthProvider` so that I can use other Microsoft clients like Microsoft Graph.

**Setup (please complete the following information):**
 - OS: OSX Big Sur
 - IDE : IntelliJ (2021.1)
 - Azure Toolkit for IntelliJ: 3.52.0-2021.1
 - aws-identity: 1.2.5 

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Bug Description Added
- [X] Repro Steps Added
- [X] Setup information Added

**Comment by joshfree** [comment]
@g2vinay could you please follow up?

**Comment by stliu** [comment]
any updates? we're running into this issue too

**Comment by andxu** [comment]
In new intellij azure toolkit releases, the plugin will use azure identity to save credentials rather than saving it at Intelij secure store(old version save the credentials) , so it is not a recommended way to use IntelliJCredential in future, in my opinion, you shall use azure cli and use:
```java
new AzureCliCredentialBuilder().tenantId(tenantId).build()
```

**Comment by billwert** [comment]
Marking this blocking so we ensure to follow up on it soon.

In the meantime, for anyone hitting this problem the best workaround is to log in with the Azure CLI and use `DefaultAzureCredential` instead of `IntelliJCredential`.

**Comment by billwert** [comment]
microsoft/azure-tools-for-java#5824 is the change that prevents `IntelliJCredential` from authenticating. This shipped in [v3.58](https://github.com/microsoft/azure-tools-for-java/releases/tag/azure-intellij-toolkit-v3.58.0) of the IntelliJ toolkit. The original issue was another problem, but now the same symptom will appear for this cause. Though it's not exactly the same, using this bug since the overall result is the same.

**Comment by joshfree** [comment]
@g2vinay can you share an update on this issue?

**Comment by joshfree** [comment]
@g2vinay could you share an update please?

**Comment by g2vinay** [comment]
The fix for this issue will be released early August with Identity release via #29527
